ForumProject/users/views.py

```diff
@@ -60,6 +66,76 @@ def post(self, request, *args, **kwargs):
         return response
 
 
+class RoleSelectionView(APIView):
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        serializer = RoleSerializer(data=request.data)
+        if serializer.is_valid():
+            user = request.user
+            if UserRoleCompany.objects.filter(user=user).exists():
+                user_role = UserRoleCompany.objects.get(user=user)
+                user_role.role = serializer.validated_data['role']
+                user_role.company_id = 0
+                user_role.save()
+            else:
+                UserRoleCompany.objects.create(user=user, role=serializer.validated_data['role'])
+            return Response({'success': 'Role has been successfully updated.'}, status=status.HTTP_200_OK)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Instead of using `UserRoleCompany.objects.filter(user=user).exists()` and then `UserRoleCompany.objects.get(user=user)`, which makes two database queries, you could use `get_or_create()` to streamline this process

Reply:
Thanks, very useful advice.